### PR TITLE
Vlad's Retrofit

### PR DESCRIPTION
**Pull request by Kryvovyaz**
8.6 HW

### CODE REVIEW

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.example.myphotos_kotlin.network.RetrofitClient
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+
+class MainActivity : AppCompatActivity() {
+    lateinit var recyclerView: RecyclerView
+
+    companion object {
```
Comment (oliver-basoglu) [inline MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt:25]: Do not create a companion object inside your Activity (View) . If KEY_PHOTO_BUNDLE is only used within MainActivity, then declare it as constant

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        val KEY_PHOTO_BUNDLE = "key_position"
+    }
+
+    lateinit var list: List<Photo>
```
Comment (oliver-basoglu) [inline MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt:29]: Always have your variables together, so move it to top under recyclerView

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            override fun onResponse(call: Call<List<Photo>>, response: Response<List<Photo>>) {
+                response.body()?.let { callPhotos ->
+                    //  list = response.body()!!
+                    CreateRecycler(callPhotos)
+
+                }
+            }
+
+            override fun onFailure(call: Call<List<Photo>>, t: Throwable) {
+                Toast.makeText(this@MainActivity, "HOUSTON WE HAVE A PROBLEM"
```
Comment (oliver-basoglu) [inline MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt:47]: Correct formatting: 
Toast.makeText(this@MainActivity, 
                         "HOUSTON WE HAVE A PROBLEM",
                          Toast.LENGTH_SHORT
                           ) .show()

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/View/PhotoAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import com.example.myphotos_kotlin.R
+import com.example.myphotos_kotlin.model.Photo
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.photo_view_holder.view.*
+
+class PhotoAdapter(val list: List<Photo>, val whattoDoOnClicks: (Int) -> Unit) :
+    RecyclerView.Adapter<PhotoAdapter.PhotoViewHolder>() {
+
+
```
Comment (oliver-basoglu) [inline MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/View/PhotoAdapter.kt:17]: Too much empty space.

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/View/PhotoAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        val view =
+            LayoutInflater.from(parent.context).inflate(R.layout.photo_view_holder, parent, false)
+        return PhotoViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: PhotoViewHolder, position: Int) {
+        holder.textViewHolder.text = list[position].title
+        Picasso.get().load(list.get(position).url).into(holder.imageViewHolder)
+        holder.itemView.setOnClickListener { whattoDoOnClicks(position) }
+
```
Comment (oliver-basoglu) [inline MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/View/PhotoAdapter.kt:28]: Remove unnecessary spaces

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/View/SinglePhotoActivity.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+class SinglePhotoActivity : AppCompatActivity() {
+    lateinit var imageView: ImageView
+    lateinit var textView: TextView
+    lateinit var photo: Photo
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_single_photo)
+        textView = findViewById(R.id.single)
+        imageView = findViewById(R.id.single_image)
+        val intenet = intent
```
Comment (oliver-basoglu) [inline MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/View/SinglePhotoActivity.kt:21]: Typo in naming. should be intent. And also remove unnecessary empty spaces

File: MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt (hunk 1 above)
Comment (Kryvovyaz) [reply MYPHOTOS_Kotlin/app/src/main/java/com/example/myphotos_kotlin/MainActivity.kt:25]: KEY_PHOTO_BUNDLE is used in SinglePhotoACTIVITY also.

**Pull request merged**